### PR TITLE
AudioRecorder

### PR DESCRIPTION
**Pull request by S-Kantor**
TODO:
- For testing, I updated the toggle button to launch the recorder. That needs to change into it's own button.
- Need to rename to AudioRecorder due to confusion
- Needs UI updates (@rohan-dave)
- Needs to send requests to the server to store the beats when the webAPI is done.

### CODE REVIEW

File: app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java [hunk 1]
```diff
@@ -310,21 +310,30 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                Intent myIntent = new Intent(CreationView.this, NotePad.class);
+                Intent myIntent = new Intent(CreationView.this, RecordingView.class);
                 startActivity(myIntent);
             }
         });
 
 
         //Setup Visualization
         AudioBar audioBar = (AudioBar) findViewById(R.id.barVisualizer);
-        audioBar.setPlayer(0);
```
Comment (S-Kantor) [inline app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: This was causing crashes on my phone.

File: app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java [hunk 2]
```diff
@@ -310,21 +310,30 @@ protected void onCreate(Bundle savedInstanceState) {
...
         Button button = findViewById(R.id.notePadLauncher);
+        toggleButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent myIntent = new Intent(CreationView.this, RecordingView.class);
+                startActivity(myIntent);
+            }
+        });
+
+        Button notePadLauncher = findViewById(R.id.notePadLauncher);
```
Comment (S-Kantor) [inline app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: This button doesn't even exist.

File: app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java (hunk 1 above)
Comment (hmtinc) [reply app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: Its cause you don't have audio permissions enabled

Comment (S-Kantor) [reply app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: 1) It should ask for it, for an example, look in this class.
2) I think I had it enabled but it crashed anyways. I can enable it and send you the logcat

Comment (hmtinc) [reply app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: Permissions is enabled during the installation process

Comment (S-Kantor) [reply app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: Then how come it wasn't when I installed the app?

**Comment by S-Kantor** [comment]
Adding another TODO: Start activity from button in Main Menu, not in CreationView

File: app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java (hunk 1 above)
Comment (hmtinc) [reply app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: I don't think the ADB connection enables permissions. Its different compared to the system app installer on android which notifies users of permissions and enables them. For development you have to manually enable permissions.

Comment (hmtinc) [reply app/src/main/java/ca/uwaterloo/cs446/teamdroids/technosync/CreationView.java]: If you uninstalled the app and installed the app via the APK, it should promote you for enabling permissions

**Pull request merged**